cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1094,7 +1096,7 @@ private KafkaVersionChange getKafkaVersionChange(StatefulSet kafkaSts) {
             return CompositeFuture.join(kafkaSetOperations.reconcile(namespace, stsName, newSts), configMapOperations.reconcile(namespace, cmName, newCm))
                     .compose(ignored -> kafkaSetOperations.maybeRollingUpdate(sts, pod -> {
                         log.info("{}: Upgrade: Maybe patch + rolling update of {}: Pod {}", reconciliation, stsName, pod.getMetadata().getName());
-                        return true;
+                        return "Upgrade: Maybe patch + rolling update of " + name + ": Pod " + pod.getMetadata().getName();
```

Comment:
There's no maybe at this point. The function is returning a non-null reason to roll, so the pod will be rolled. The interesting information is that the rolling is due to upgrade.